src/test/scala/mesosphere/marathon/integration/setup/ProcessOutputToLogStream.scala

```diff
@@ -1,13 +1,40 @@
 package mesosphere.marathon
 package integration.setup
 
-import org.slf4j.LoggerFactory
+import org.slf4j.{ LoggerFactory, Logger }
 
 import scala.sys.process.ProcessLogger
 
-case class ProcessOutputToLogStream(process: String) extends ProcessLogger {
-  val log = LoggerFactory.getLogger(s"mesosphere.marathon.integration.process.$process")
-  override def out(msg: => String): Unit = log.debug(msg)
-  override def err(msg: => String): Unit = log.warn(msg)
+final case class ProcessOutputToLogStream(process: String, maxLinesPerPeriod: Int = 100, periodMillis: Long = 1000L) extends ProcessLogger {
+  private[this] var nextFlush = System.currentTimeMillis() + periodMillis
+  private[this] var callsSinceLastFlush = 0
+  val log: Logger = LoggerFactory.getLogger(s"mesosphere.marathon.integration.process.$process")
+
+  private def rateLimited(msg: String): Boolean = synchronized {
```

Comment:
Unfortunately I have a hard time understanding the logic.
 * When do we flush? I do not see any call to flush.
 * In both if-branches we might rate limit, right? So why do we log only once that we rate limit?

Also, shouldn't this method be threadsafe? By changing `callsSinceLastFlush` it does not seem to be.

Reply:
On second thought, flush was a bad choice in naming; probably better to use reset.

How is the method not threadsafe? It is wrapped in a synchronized block.

Reply:
Oh, I've missed the `synchronized`.